invoke-ai/InvokeAI
# feat(ui): add LoRA support to SDXL linear UI

## What type of PR is this? (check all applicable)


## Have you discussed this change with the InvokeAI team?

## Description

Adds SDXL LoRA support in the Linear UI.

- new graph modifier `addSDXLLoRasToGraph()` handles adding LoRA to the SDXL t2i and i2i graphs.

## QA Instructions, Screenshots, Recordings

1. Download and install https://civitai.com/models/118653?modelVersionId=128752 (per author, this only modifies UNet, and needs to be used with weight 2)
2. Restart server
3. Navigate to t2i tab
4. Write a simple prompt with `oil painting` in the name with manual seed, generate
5. Add the LoRA with weight 2, generate
6. Confirm the results are much more oil-painting-y
7. Generate an image without `oil painting` in the prompt, without LoRAs, to be used as init image
8. Navigate to i2i tab
9. Select the image from step 7 as init image
10. Add `oil painting` to prompt, change seed (to avoid fried images), generate
11. Add the LoRA with weight 2, generate
12. Confirm the LoRA'd result is much more oil-painting-y than the non-LoRA'd result

### Expected Outputs

t2i without LoRA
![image](https://github.com/invoke-ai/InvokeAI/assets/4822129/1e7d3fa5-f1c1-4c9b-aa8e-859e9f61ebd3)

t2i with LoRA
![image](https://github.com/invoke-ai/InvokeAI/assets/4822129/c7cde244-73d5-4bbc-a19d-496161091fde)

i2i init image
![image](https://github.com/invoke-ai/InvokeAI/assets/4822129/7183051e-b7a6-41a8-93a5-e934527e4964)

i2i without LoRA
![image](https://github.com/invoke-ai/InvokeAI/assets/4822129/d5c1710d-a93b-4574-b530-069e0aeef8a2)

i2i with LoRA
![image](https://github.com/invoke-ai/InvokeAI/assets/4822129/7df3aac0-9ba5-42cd-9228-23e96c51394a)

(freaking *awesome* LoRA btw, LOOK AT THOSE KITTIES)
